File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    rightSideView(root) {
        // inorder
        let res = [];
        let que = [];
        if (root) {
            que.push(root);
        }
        while ( que.length > 0 ) {
            let length = que.length;
            for (let i = 0; i < length; i++ ){
                var curr = que.shift();
                if (curr.left){
                    que.push(curr.left);
                }
                if (curr.right){
                    que.push(curr.right);
                }
            }
            res.push( curr.val )
        }
    return res
    }
}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        def dfs(root):
            if not root:
                return
            if root.val == q.val or root.val == p.val:
                return root
            left = dfs(root.left)
            right = dfs(root.right)
            if left and right:
                return root
            return left or right
        return dfs(root)